### PR TITLE
fix(autosuggestbox): bring selected item from popup into view + resize popup to match the size of the ASB changes

### LINKED ISSUE #11876: [WASM/Skia] `AutoSuggestBox` scroll doesn't follow selected item
### Current behavior

When using arrows to navigate through AutoSuggestBox's suggested items the scroll does not follow the selected item. 

https://user-images.githubusercontent.com/78549750/229382764-943c4b0f-2440-4b73-8f5c-99fd3dd2632a.mp4

### Expected behavior

The scroll should follow the selected item as it does on UWP.

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

WebAssembly, Skia

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### LINKED ISSUE #17907: [Skia] `AutoSuggestBox` flyout doesn't update width
### Current behavior

https://github.com/user-attachments/assets/1a1cce31-220a-4be5-ac62-27061a285646

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

"Uno.Sdk": "5.3.96" or "Uno.Sdk": "5.4.0-dev.180"

### Affected platforms

Skia (WPF), Skia (Linux X11), Skia (macOS), Skia (Linux Framebuffer), Skia (GTK)

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #11876, closes #17907

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 1]
```diff
@@ -333,6 +337,11 @@ private void OnPopupOpened(object sender, object e)
 		{
 			LayoutPopup();
 		}
+g
+		private void OnSizeChanged(object sender, SizeChangedEventArgs args)
+		{
+			LayoutPopup();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:343]: I think the correct thing is to set the Width on `SuggestionsContainer` to match WinUI

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:343]: Relevant WinUI code:

```cpp
//------------------------------------------------------------------------------
// Handler of the SizeChanged event.
//------------------------------------------------------------------------------
_Check_return_ HRESULT
AutoSuggestBox::OnSizeChanged(
    _In_ IInspectable* /*pSender*/,
    _In_ xaml::ISizeChangedEventArgs* /*pArgs*/)
{
    HRESULT hr = S_OK;

    if (m_tpSuggestionsContainerPart)
    {
        DOUBLE actualWidth = 0.;
        wrl::ComPtr<xaml::IFrameworkElement> spThisAsFE = this;

        IFC(spThisAsFE->get_ActualWidth(&actualWidth));
        IFC(m_tpSuggestionsContainerPart->put_Width(actualWidth));
    }

Cleanup:
    RRETURN(hr);
}

```

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 2]
```diff
@@ -337,10 +337,13 @@ private void OnPopupOpened(object sender, object e)
...
 			LayoutPopup();
 		}
-g
+
 		private void OnSizeChanged(object sender, SizeChangedEventArgs args)
 		{
-			LayoutPopup();
+			if (_suggestionsList is not null)
+			{
+				_suggestionsList.Width = ActualWidth;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:345]: It should be `_suggestionsContainer` I think?

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 3]
```diff
@@ -444,6 +456,7 @@ private void HandleUpDownKeys(KeyRoutedEventArgs e)
 			}
 
 			_suggestionsList.SelectedIndex = nextIndex;
+			_suggestionsList.ScrollIntoView(_suggestionsList.SelectedItem);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:459]: It feels like this shouldn't be specific to up/down keys, but should happen whenever `_suggestionsList.SelectedItem` changes? i.e, listen to `_suggestionsList.SelectionChanged`?

**Review by Youssef1313 — APPROVED** [review]
ASB changes LGTM, with minor comment regarding listening to the suggestion list selection change.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs (hunk 3 above)
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:459]: Good catch. I tested on WinUI and it indeed scrolling into view even with an explicit call to ListView.set_SelectedIndex.

**Pull request merged**